lib/adroll/api/ad.rb

```diff
@@ -3,38 +3,42 @@
 module AdRoll
   module Api
     class Ad < AdRoll::Api::Service
-      WHITELIST_PARAMS = [:ad, :advertisable, :name, :destination_url, :file,
-                          :headline, :body, :message, :headline_dynamic,
-                          :body_dynamic, :message_dynamic, :is_fb_dynamic,
-                          :dynamic_template_id, :product, :logo, :background,
-                          :call_to_action, :multiple_products, :tracking,
-                          :lead_gen_form_id, :multi_share_optimized, :child_ads,
-                          :app_id, :ad_format, :prefix, :display_url_override, :type]
+      WHITELIST_PARAMS = [:ad, :ad_format, :advertisable, :app_id, :background,
+                          :body, :body_dynamic, :call_to_action, :child_ads,
+                          :destination_url, :display_url_override,
+                          :dynamic_template_id, :file, :headline,
+                          :headline_dynamic, :is_fb_dynamic, :lead_gen_form_id,
+                          :logo, :message, :message_dynamic,
+                          :multi_share_optimized, :multiple_products, :name,
+                          :prefix, :product, :tracking, :type].freeze
 
       def clone(params)
-        call_api(:post, __method__, validate_params(params))
+        call_api(:post, __method__, sanitize_params(params))
       end
 
       def create(params)
-        call_api(:post, __method__, validate_params(params))
+        call_api(:post, __method__, sanitize_params(params))
       end
 
       def edit(params)
-        call_api(:put, __method__, validate_params(params))
+        call_api(:put, __method__, sanitize_params(params))
       end
 
       def get(params)
-        call_api(:get, __method__, validate_params(params))
+        call_api(:get, __method__, sanitize_params(params))
       end
 
+      # undocumented
       def set_outline(params)
-        call_api(:get, __method__, validate_params(params))
+        call_api(:get, __method__, sanitize_params(params))
       end
 
       private
 
-      def validate_params(params)
-        params.reject { |key, value| !WHITELIST_PARAMS.include?(key) || value.nil? }
+      def sanitize_params(params)
```

Comment:
The sanitize_params method is the same in all the services so let's move it to the AdRoll::Api::Service and AdRoll::Uhura::Service